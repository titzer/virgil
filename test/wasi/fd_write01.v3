def iovecs = Array<int>.new(4);
def sizebuf = Array<int>.new(1);

def main(a: Array<string>) -> int {
	var msg = "fd_write: success\n";

	var ioptr = Pointer.atContents(iovecs);
	var sizeptr = Pointer.atContents(sizebuf);
	
	for (i < msg.length) {
		iovecs[0] = Pointer.atElement(msg, 0) - Pointer.NULL;
		iovecs[1] = i;
		iovecs[2] = Pointer.atElement(msg, i) - Pointer.NULL;
		iovecs[3] = msg.length - i;

		var r = wasi_snapshot_preview1.fd_write(1, ioptr, 2, sizeptr);
		if (r != 0) return -1 - sizebuf[0];
	}
	return 43;
}
