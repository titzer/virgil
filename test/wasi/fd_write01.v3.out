fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
fd_write: success
